docs/architecture/accounts.md

```diff
@@ -1,211 +1,139 @@
 
-An account vault can contain an unlimited number of [assets](assets.md). The assets are stored in a sparse Merkle tree as follows:
+- **Faucets:**  
+  Faucets are always immutable and can be specialized by the type of assets they issue:
+  - *Fungible Faucet:* Can issue fungible [assets](assets.md).
+  - *Non-fungible Faucet:* Can issue non-fungible [assets](assets.md).
 
-* For fungible assets, the index of a node is defined by the issuing faucet ID, and the value of the node is the asset itself. Thus, for any fungible asset, there will be only one node in the tree.
-* For non-fungible assets, the index is defined by the asset itself, and the asset is also the value of the node.
+Type and mutability are encoded in the two most significant bits of the account's ID:
 
-An account vault can be reduced to a single hash which is the root of the sparse Merkle tree.
-
-### Code
-
-The interface for accounts. In Miden, every account is a smart contract. It has an interface that exposes functions that can be called by [note scripts](notes.md#the-note-script) and transaction scripts. Users cannot call those functions directly.
-
-Functions exposed by the account have the following properties:
-
-* Functions are actually roots of [Miden program MASTs](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html) (i.e., a `32`-byte hash). Thus, the function identifier is a commitment to the code which is executed when a function is invoked.
-* Only account functions have [mutable access](transactions/contexts.md) to an account's storage and vault. Therefore, the only way to modify an account's internal state is through one of the account's functions.
-* Account functions can take parameters and can create new notes.
-
-> **Note**
-> Since code in Miden is expressed as MAST, every function is a commitment to the underlying code. The code cannot change unnoticed to the user because its hash would change. Behind any MAST root there can only be `256` functions.
-
-#### Example account code
+|                        | Basic Mutable | Basic Immutable | Fungible Faucet | Non-fungible Faucet |
+|------------------------|---------------|-----------------|-----------------|---------------------|
+| **Description**        | For general user accounts (e.g., a wallet) where code changes are allowed. | For typical smart contracts where code is fixed after deployment. | Allows to issue and manage fungible assets. | Allows to issue and manage non-fungible assets. |
+| **Code updatability**  | Yes           | No              | No              | No                  |
+| **Most significant bits** | `00`        | `01`            | `10`            | `11`                |
 
-Currently, Miden provides two standard implementations for account code.
+### The Accounts Storage Mode
 
-##### Basic user account
+Users can choose whether their accounts are stored publicly or privately. The third and fourth most significant bits of the [ID](#id) encode this preference:
```

Comment:
Because we explain the bit logic in the account ID already, I would just say here: 

"the preference is encoded in the accounts ID" or something like that